### DISCUSSION TITLE
Centre stimulus vertically

### DISCUSSION THREAD

**Discussion by Jessica-Schulz** [jspsych/jsPsych#2097] [Q&A]
Hello!

I am using the html-keyboard-response plugin to show a series of stimuli (single words) following this pattern: fixation cross --> blank screen --> stimulus

I would like to include a prompt (reminder of response keys) that appears _below_ the stimulus, whilst the stimulus itself remains centered both vertically and horizontally _in the same position as where the fixation cross was_. However, when I include a prompt using the prompt parameter, the stimulus appears above the position of the prior fixation cross and the prompt appears below it. In other words, stimulus and prompt are centered _as one unit_.

How do I make the stimulus appear in the centre of the screen where the fixation cross was, and the prompt below it? I have tried different things using css and html code, but I haven't been able to find a solution so far.

Any ideas would be greatly appreciated!

**Comment by kurokida** [comment]
Hi, @Jessica-Schulz ,

This is not a solution for the html-keyboard-response plugin, but a plugin I created may be useful. Please see [the psychophysics plugin](https://jspsychophysics.hes.kyushu-u.ac.jp/).

**Reply by Jessica-Schulz** [reply]
Hi @kurokida,

Thank you very much for your help! I used your plugin and it does exactly what I want it to do!

The text on the canvas shows up a bit blurry though - is there a way to achieve the same text sharpness as with html?

Thank you!

**Reply by kurokida** [reply]
Thank you for using my plugin!

I've never seen the blurring. Could you show me the code about the text object?
I think this problem is related to the font settings.

One more thing, please let me know what the text of [this demo program](https://www.hes.kyushu-u.ac.jp/~kurokid/jspsychophysics/demos/texts.html) looks like.

**Reply by Jessica-Schulz** [reply]
Thank you for your reply! Here is a screenshot of the code for one of my text objects - I have specified the font for the other text objects in a similar way:

<img width="228" alt="Screenshot 2021-08-27 at 16 30 09" src="https://user-images.githubusercontent.com/89267642/131152714-6f0a2174-f482-4880-9ed7-e9778e5b1e43.png">

This results in:

<img width="499" alt="Screenshot 2021-08-27 at 16 19 40" src="https://user-images.githubusercontent.com/89267642/131152750-8bf97a9f-cc22-4f0f-9d00-9752a81c1d53.png">

which I find more blurry than:

<img width="502" alt="Screenshot 2021-08-27 at 16 24 53" src="https://user-images.githubusercontent.com/89267642/131152773-8cf38e90-ec1f-40bd-b8db-71b067022235.png">

(The difference becomes more apparent on a larger screen, such as a laptop.)

The text of the demo you sent me looks a bit blurry to me too.

**Reply by kurokida** [reply]
Thank you for your reply.

I probably found a solution for this problem.
Perhaps this problem is related to [the devicePixelRatio](https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio).

I've updated my plugin in a new branch named as [blur_problem](https://github.com/kurokida/jspsych-psychophysics/tree/blur_problem).

You can download the source and sample code from this branch. Please be careful to change the branch from master to blur_problem.

This is [the sample page](https://www.hes.kyushu-u.ac.jp/~kurokid/blur/demos/retina_display.html).

In a normal display, the devicePixelRatio retruns 1, while in a retina (high resolution) display you are using, it returns 2.
Could you check the value in your display?

Please note that a variety of displays will be used in online experiments. I believe this update will fix the problem, but you should check it works on a variety of displays.

**Reply by Jessica-Schulz** [reply]
Thank you very much for your help and for solving this problem for me! I checked the devicePixelRatio value on my display and it is indeed 2. The text is sharp now with your fix, and I will check to make sure that it works on other displays too!

Thank you very much!
